### DISCUSSION TITLE
Lesson 6: API Key issues

### DISCUSSION THREAD

**Discussion by ArcusConsulting** [smartcontractkit/full-blockchain-solidity-course-py#257] [Q&A] [answered]
I generated my API key but my terminal says I have an invalid API key:

Windows 10 machine
Python 3.9.0

deploy.py
![image](https://user-images.githubusercontent.com/91853555/138619424-ace63dff-bf5f-4551-9364-f460a0eb1a8e.png)

helpful_scripts.py
![image](https://user-images.githubusercontent.com/91853555/138619446-369aa72a-af8f-4bcb-a3cd-cbc799ca0485.png)

.env
![image](https://user-images.githubusercontent.com/91853555/138619484-0b02311c-ef7f-497a-886f-a992221b8f06.png)

etherscan.io
![image](https://user-images.githubusercontent.com/91853555/138619503-f6572a98-f70a-4a26-bee1-81ae2b667f87.png)

brownie-config.yaml
![image](https://user-images.githubusercontent.com/91853555/138619525-ff002c23-2d1d-4100-962e-b572ed47069a.png)

File tree:
![image](https://user-images.githubusercontent.com/91853555/138619538-9226faaa-8205-489f-9d18-5b93dfa88922.png)

Terminal:
![image](https://user-images.githubusercontent.com/91853555/138619577-f3e9184c-8b2d-4495-9b99-00e35ca1e39a.png)

**Comment by ArcusConsulting** [comment]
Looking forward to @PatrickAlphaC's  help, as always!!

**Comment by cromewar** [comment, marked as answer]
Hey @ArcusConsulting I've reviewed your code and I didn't find anything wrong, however I have some suggestions for you.

1. Make sure your private key correspond to your Rinkeby address.
2. Sometimes brownie does not update the `.env` and you have to add those environment variables manually, so:
   - Close the project.
   - Open it again.
   - Delete the build folder
   - compile with: brownie compile
   - try to deploy again.
   This should load all the `.env` file again.
3. if nothing works, add the API key as environment variable manually through windows advanced configuration.

Final hint: Did you ever tried using WSL (windows subsystem for linux)? this would make your life easier, as if you want no add a new env variable to your path you just have to use `export VARIABLE=value`

**Reply by ArcusConsulting** [reply]
Awesome... i did your Step 2 and it worked:
![image](https://user-images.githubusercontent.com/91853555/138745441-d85bab85-d244-4b17-9353-9e7a325fffaf.png)

I'm a fish out of water with this stuff... I tried learning Python for the last 3 year and nothing has stuck. I use a few scripts for work (just deleting & combining PDFs) and am quite proficient with SQL, but this is a whole new world for me. I don't understand most of this 16h video, however I am fascinated by the Blockchain space and want to acquire a (future) highly in-demand skill. Thank you for taking the time to read over my code and having the patience to answer my questions.

**Reply by cromewar** [reply]
Well being honest I also have just one month or so learning blockchain technologies and that's why I decided to help the community because all of this could be overwhelming so I'm glad my answers are helping you!

**Reply by Anu-srinivasan** [reply]
@cromewar 

"if nothing works, add the API key as environment variable manually through windows advanced configuration."
I have added the infura ID and private key manually in user environment variable manually through windows advanced configuration. However, I am clueless how to fetch these in FundMe -Brownie lesson. 
It works in local ganache. It is not getting deployed for testnet such as Rinkeby. Let me know how to load both variables to Eth-brownie project from advanced configuration please

**Reply by PatrickAlphaC** [reply]
@Anu-srinivasan, could you make a new discussion/ thread with as much detailed information as possible?

And yes, @cromewar is awesome :)

**Reply by cromewar** [reply]
@Anu-srinivasan  if you key is now on you OS variables you should use `os.getenv()` please refer to [this](https://www.geeksforgeeks.org/python-os-getenv-method/)

And thanks @PatrickAlphaC
